src/baldr/graphreader.cc

```diff
@@ -340,13 +340,18 @@ TileCache* TileCacheFactory::createTileCache(const boost::property_tree::ptree&
 }
 
 // Constructor using separate tile files
-GraphReader::GraphReader(const boost::property_tree::ptree& pt)
+GraphReader::GraphReader(const boost::property_tree::ptree& pt,
+                         std::unique_ptr<tile_getter_t>&& tile_getter)
     : tile_extract_(get_extract_instance(pt)), tile_dir_(pt.get<std::string>("tile_dir", "")),
-      curlers_(std::make_unique<curler_pool_t>(pt.get<size_t>("max_concurrent_reader_users", 1),
-                                               pt.get<std::string>("user_agent", ""))),
-      tile_url_(pt.get<std::string>("tile_url", "")),
-      tile_url_gz_(pt.get<bool>("tile_url_gz", false)),
-      cache_(TileCacheFactory::createTileCache(pt)) {
+      tile_getter_(std::move(tile_getter)),
```

Comment:
its already an r-value reference, do we really need to `move` it again?

Reply:
I guess we do since every rvalue is an lvalue inside the scope of the function until we _cast_ it to an rvalue.

Reply:
oh right! this feels like a wart of the language sorry for the noise